src/extract.rs

```diff
@@ -11,13 +11,47 @@ use futures_core::ready;
 
 use crate::{dev::Payload, Error, HttpRequest};
 
-/// Trait implemented by types that can be extracted from request.
+/// [`FromRequest`] is one of the most commonly used trait, although some may not realize it.
 ///
-/// Types that implement this trait can be used with `Route` handlers.
+/// A type that implements [`FromRequest`] is called an **extractor** and can be extracted
+/// from the request, for example [`Json`], [`Form`], [`Path`] and so on.
+///
+/// An extractor can be customized by injecting the corresponding configuration with:
```

Comment:
I'd suggest putting this part below the list of the built-in extractors, so the reader can actually look at the configs, and then see how he can inject their own.